dist/angular-socialshare.js

```diff
@@ -514,6 +514,12 @@
           + ',top=' + ($window.innerHeight - attrs.socialsharePopupHeight) / 2 + ',left=' + ($window.innerWidth - attrs.socialsharePopupWidth) / 2);
       }
     }
+    , facebookMessengerShare = function facebookMessengerShare($window, $location, attrs, element) {
+
+      var href = 'fb-messenger://share?link=' + encodeURIComponent(attrs.socialshareUrl || $location.absUrl());
+
+      element.attr('href', href);
```

Comment:
Not even this works, i tried with an hidden <a> and triggering click() on it doesn't work, seems really like it only works if it's an href="" element only and visible in page :(